sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/AlterTableExec.scala

```diff
@@ -44,3 +45,34 @@ case class AlterTableExec(
     Seq.empty
   }
 }
+
+/**
+ * Physical plan node for adding a check constraint with validation.
+ */
+case class AddCheckConstraintExec(
+    catalog: TableCatalog,
+    ident: Identifier,
+    changes: Seq[TableChange],
+    condition: String,
+    child: SparkPlan) extends V2CommandExec with UnaryExecNode {
+
+  override def output: Seq[Attribute] = Seq.empty
+
+  override protected def run(): Seq[InternalRow] = {
+    try {
+     if (child.executeTake(1).nonEmpty) {
+       throw QueryExecutionErrors.newCheckViolation(condition, ident.name())
```

Comment:
shall we report the qualified name in the error message? it should be `ident.toQualifiedNameParts(catalog)` (need to import `CatalogV2Implicits`), and in `def newCheckViolation`, we call `toSqlId` to generate qualified table name from `Seq[String]`